### PR TITLE
TECH-863: fixes to gherkin and js tests

### PR DESCRIPTION
**Pull request by lruzicki**
**Ticket:**
https://govstack-global.atlassian.net/browse/TECH-863

**Changes:**
1. Lack of content type as a string - it should be a string to automatically generate JS tests.
   - Correction: "application/json"

2. The request body is in the format application/x-www-form-urlencoded, which is not used elsewhere.
   - Correction: The POST request body has a content type of "application/x-www-form-urlencoded."

3. The body parameters of the endpoint are in snake case, not camel case as we have now.
   - Correction: Camel case standard for body key names.

4. Missing required body parameters in the positive response.
   - Missing parameter: id_token
   - Correction: "id_token": "string", "access_token": "string", "token_type": "Bearer", "expires_in": 0

5. Missing tags.
   - Missing tags: @positive + @unit

6. Lack of assertion for positive tests in the response.
   - Correction: Include assertions with the set parameters.

7. No information about the REQUIRED header.
   - Missing header: Header X-XSRF-TOKEN is required.

8. RequestTime is required in the request body.
   - Correction: "requestTime": "string"

9. Incorrect parameter name: linkCode
   - Correction: Change to linkedCode

10. Spelling error: reaused completed transactionId
    - Correction: Change to already used completed transactionId

11. Missing parameter - written as separate words.
    - Correction: Change "link code connected to a different transactionId" to "linkCode connected to a different transactionId"
    
**Testing:**
All updated tests are functioning as expected. One test was not working initially, and it remains unresolved. Addressing this issue is outside the scope of this PR and will require further investigation.

### CODE REVIEW

**Review by dborowiecki — APPROVED** [review]

**Comment by dborowiecki** [comment]
@lruzicki while I approved the changes, not all tests seem to be passing for the mocked instance. Please check the output: https://app.circleci.com/pipelines/github/GovStackWorkingGroup/bb-identity/184/workflows/149107a4-4070-474f-bb04-c9f747e23999/jobs/447/tests and let me know if any changes to code are required.

**Comment by lruzicki** [comment]
> @lruzicki while I approved the changes, not all tests seem to be passing for the mocked instance. Please check the output: https://app.circleci.com/pipelines/github/GovStackWorkingGroup/bb-identity/184/workflows/149107a4-4070-474f-bb04-c9f747e23999/jobs/447/tests and let me know if any changes to code are required.

I think we can skip it and resolve this issue in another PR since we need more information about how it has to be handled and why this test was not completed

File: test/openAPI/features/oauth_token.feature [hunk 1]
```diff
@@ -4,24 +4,30 @@ Feature: API to create access token
...
+    When The User sends POST request with given "authorization_code" as grantType, "code" as code, "<clientId>" as clientId, "urn:ietf:params:oauth:client-assertion-type:jwt-bearer" as clientAssertionType, "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9." as clientAssertion, "<redirectUri>" as redirectUri, "<clientId>" as iss, "<clientId>" as sub, "<audience>" as aud, 1024 as exp, 1516239022 as iat, "content-type": "application/x-www-form-urlencoded" header
     Then User receives a response from the POST /oauth/token endpoint
     And The POST /oauth/token endpoint response should be returned in a timely manner 15000ms
     And The POST /oauth/token endpoint response should have status 200
-    And The POST /oauth/token endpoint response should have content-type: application/json header
+    And The POST /oauth/token response should have "content-type": "application/json" header
     And The POST /oauth/token endpoint response should match json schema
     And The POST /oauth/token endpoint response should contain "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9" as idToken
+    And The POST /oauth/token endpoint response should contain "string.string.string" as accessToken
```
Comment (dborowiecki) [inline test/openAPI/features/oauth_token.feature:28]: Shouldn't we just check if "accessToken" is accessible?

**Review by dborowiecki — CHANGES_REQUESTED** [review]
@lruzicki I left one comment. I'm not entirely sure what approach we should take so please find my comment.

File: test/openAPI/features/oauth_token.feature (hunk 1 above)
Comment (lruzicki) [reply test/openAPI/features/oauth_token.feature:28]: yeah I think so, I will change it to check for parameter only

**Review by dborowiecki — APPROVED** [review]

**Pull request merged**